warehouse/internal/repo/staging.go

```diff
@@ -251,22 +251,42 @@ func (repo *StagingFiles) GetByID(ctx context.Context, ID int64) (model.StagingF
 	return *entries[0], err
 }
 
-// GetSchemaByID returns staging file schema field the given ID.
-func (repo *StagingFiles) GetSchemaByID(ctx context.Context, ID int64) (jsonstd.RawMessage, error) {
-	query := `SELECT schema FROM ` + stagingTableName + ` WHERE id = $1`
+// GetSchemasByIDs returns staging file schemas for the given IDs.
+func (repo *StagingFiles) GetSchemasByIDs(ctx context.Context, ids []int64) ([]model.Schema, error) {
+	query := `SELECT schema FROM ` + stagingTableName + ` WHERE id = ANY ($1);`
 
-	row := repo.db.QueryRowContext(ctx, query, ID)
-	if row.Err() != nil {
-		return nil, fmt.Errorf("querying staging files: %w", row.Err())
+	rows, err := repo.db.QueryContext(ctx, query, pq.Array(ids))
+	if err != nil {
+		return nil, fmt.Errorf("querying schemas: %w", err)
 	}
+	defer func() { _ = rows.Close() }()
 
-	var schema jsonstd.RawMessage
-	err := row.Scan(&schema)
-	if err != nil {
-		return nil, fmt.Errorf("parsing rows: %w", err)
+	schemas := make([]model.Schema, 0, len(ids))
+
+	for rows.Next() {
+		var (
+			rawSchema jsonstd.RawMessage
+			schema    model.Schema
+		)
+
+		if err := rows.Scan(&rawSchema); err != nil {
+			return nil, fmt.Errorf("scanning row: %w", err)
```

Comment:
We might be missing some context here, perhaps it makes sense to prepend the schema function in order to ease debug when coming across these errors:
```suggestion
			return nil, fmt.Errorf("cannot get schemas by ids: scanning row: %w", err)
```

Reply:
The same would apply to other errors like the ones below e.g. `fmt.Errorf("iterating rows: %w", err)` etc...